config/samples/httpbin_certificate_pinning.yaml

```diff
@@ -0,0 +1,65 @@
+# Certificate Pinning
+#
+# Certificate pinning is a feature which allows you to whitelist public keys used to generate certificates. For more
+# details, please visit official Tyk Documentation https://tyk.io/docs/security/certificate-pinning/.
+#
+# Public keys can be defined in two ways:
+# 1) Through ApiDefinition object's `pinned_public_keys` field, using the following format:
+#  pinned_public_keys:
+#    "foo.com": "<key_id>",
+#    "*": "<key_id-1>,<key_id-2>"
+#
+# Public keys are stored inside the Tyk certificate storage, so you can use Certificate API to manage them. For key-id
+# you should set the ID returned after you upload the public key using the Certificate API.
+#
+# 2) Through ApiDefinition object's 'pinnedPublicKeysSecretNames' field, using the following format:
+#  spec:
+#   pinnedPublicKeysSecretNames:
+#    "domain.org": <secret_name> # the name of the Kubernetes Secret Object that holds the public key for the 'domain.org'.
+#
+# In this way, you can refer Kubernetes Secret Objects through 'pinnedPublicKeysSecretNames' field.
+#
+# NOTE: For now, only Opaque type of Secret Objects are allowed. Also, please use 'public-key' field in your Secret object's
+#   data field (see the below example). Secret-Cert controller and ApiDefinition controller will check 'public-key' field
+#   of your Opaque secret for the public key.
+# NOTE: The secret that includes a public key must be in the same namespace as the ApiDefinition.
+#
+# In this example, we have an HTTPS upstream target as `https://httpbin.org`. The public key of httpbin.org is obtained
+# with the following command:
+#   $ openssl s_client -connect httpbin.org:443 -servername httpbin.org 2>/dev/null | openssl x509 -pubkey -noout
+#
+# We are creating a secret called 'httpbin-secret'. In the 'public-key' field, we are specifying the public key of the
+#  httpbin.org obtained with above `openssl` command, in the decoded manner.
+# Now, in the ApiDefinition, we are using this public key for all domains as '*' (wildcard domains are also supported).
+# If you target any other URL than https://httpbin.org (for instance https://github.com/), you will face a 'public key pinning error'
+# for that particular domain because the public key of httpbin.org is used for all domains.
+apiVersion: v1
+kind: Secret
+metadata:
+  name: httpbin-secret
+type: Opaque
+data:
+  public-key: | # This field name is fixed and shouldn't be changed. Secret controller strictly checks for `public-key` field.
```

Comment:
Is that a good design idea?

Reply:
@asoorm Isn't tls secret used to store private key and certificate?

Reply:
For future reference, we decided to use the tls-secret type instead of Opaque